Feature 1230 stratosphere metrics

hankenstein2 opened the pull request:
Stratosphere metrics updates

Commit:
Initial checkin for Meridial Mean use case

Commit:
Merge branch 'feature_1230_stratosphere_metrics' of ssh://github.com/dtcenter/METplus into feature_1230_stratosphere_metrics

hankenstein2 committed:
Changed the name and directories

hankenstein2 committed:
Added some documentation

Commit:
Fixed config variables

Commit:
Merge branch 'develop' of https://github.com/dtcenter/METplus into feature_1230_stratosphere_metrics

Commit:
Issue 1230 Stratospheric metrics (zonal/meridional use case) put INPUT_BASE in the user_env_vars so the user doesn't need to set the INPUT_BASE environment in the current working shell

Commit:
Issue #1230 remove import of metplotpy-this isn't called

Commit:
Issue #1230_stratosphere_metric Add use case to the list of use cases to be run for testing

Commit:
Issue 1230 Remove the INPUT_BASE from the user_env_vars section

Commit:
Issue #1230 redundant files

Commit:
Issue #1230 redundant files

Commit:
Issue #1230 redundant files

Commit:
Issue #120 replace INPUT_BASE with INPUT_FILE_NAME

Commit:
Issue #1230 remove entry for INPUT_FILE_NAME, this goes in the system.conf

georgemccabe commented:
Closing PR so full automation test suite doesn't run for each commit. Re-open when ready for review.

georgemccabe commented on internal_tests/use_cases/all_use_cases.txt:
```diff
@@ -129,6 +129,7 @@ Category: s2s
 9:: UserScript_obsERA_obsOnly_OMI:: model_applications/s2s/UserScript_obsERA_obsOnly_OMI.conf:: spacetime_env, metdatadb
 10:: UserScript_obsERA_obsOnly_RMM:: model_applications/s2s/UserScript_obsERA_obsOnly_RMM.conf:: spacetime_env, metdatadb
 11:: UserScript_fcstGFS_obsERA_WeatherRegime:: model_applications/s2s/UserScript_fcstGFS_obsERA_WeatherRegime.conf:: weatherregime_env,cartopy,metplus
+11:: UserScript_obsERA_obsOnly_Stratosphere:: model_applications/s2s/UserScript_fcstGFS_obsERA_Stratosphere.conf:: metplotpy_env,metplus
```
There are 2 entries under s2s that have index 11. Change the 2nd one to 12. Also, the use case will not run in the automated test suite unless it is added as a new group to .github/parm/use_case_groups.json:

> 
>   {
>     "category": "s2s",
>     "index_list": "12",
>     "run": true
>   },

Setting run to true will run this use case group for each push to the branch. This can be used for testing the new use case. When the use case is running without issues, change run to false so that it does not run for other feature branches. More info here: https://metplus.readthedocs.io/en/develop/Contributors_Guide/add_use_case.html#add-new-category-to-test-runs

georgemccabe commented on parm/use_cases/model_applications/s2s/UserScript_obsERA_obsOnly_Stratosphere/meridonial_mean.yaml:
```diff
@@ -0,0 +1,2 @@
+input_filename: 
+- !ENV '${INPUT_FILE_NAME}/model_applications/s2s/UserScript_obsERA_obsOnly_Stratosphere/SSWC_v1.0_varFull_ERAi_d20130106_s20121107_e20130307_c20160701.nc'
```
The input for the use case needs to be added to the directory with the same name as this feature branch on the DTC web server to make it available in the automated tests. Detailed steps to add new data is here: https://metplus.readthedocs.io/en/develop/Contributors_Guide/add_use_case.html#input-data
**Be sure to read each step carefully! Some instructions are only run on certain conditions.**

bikegeek committed:
Update all_use_cases.txt

Copy and pasted #11 from s2s use case for the Stratosphere use case but forgot to update the index to 12.

bikegeek committed:
Update use_case_groups.json

added use case #12 from s2s to test stratosphere use case

Commit:
Issue #1230 put INPUT_FILE_NAME back under the user_env_vars

Commit:
Merge branch 'feature_1230_stratosphere_metrics' of https://github.com/dtcenter/METplus into feature_1230_stratosphere_metrics

Commit:
Issue #1230 forgot to include the filename

Commit:
Issue #1230 clean up config file, remove uneccessary comments, group related entries in the user_env_vars

Commit:
Issue #1230 remove extraneous and incorrect path to the input_filename setting

Commit:
Removed pingouin dependency

Commit:
Issue #1230 another cut and paste error fixed for the Stratosphere use case

Commit:
Issue #1230 type in name, obs_Only should be obsOnly

Commit:
Issue #1230 added the metdatadb to the env, code imports metdatadb and may require some of these dependencies

Commit:
issue #1230 removed pingouin dependency from comment to reduce any confusion

Commit:
issue #1230 turn off the test for the Stratosphere metrics use case

bikegeek reviewed (changes requested):
Code is behaving as expected.  Need documentation for the use case.

CPKalb commented:
I can add some documentation, but it will probably be later tonight or tomorrow morning when I get to it.

Commit:
Added use case image
